Sources/TensorFlow/Core/Tensor.swift

```diff
@@ -239,11 +246,14 @@ public extension Tensor {
     /// - Parameters:
     ///   - shape: The shape of the tensor.
     ///   - scalars: The scalar contents of the tensor.
-    /// - Precondition: The number of scalars must equal the product of the
-    ///   dimensions of the shape.
+    /// - Precondition: The product of the dimensions of the shape must equal the number of scalars.
     @inlinable
     init(shape: TensorShape, scalars: UnsafeBufferPointer<Scalar>) {
-        precondition(scalars.count == shape.contiguousSize)
+        precondition(shape.contiguousSize == scalars.count,
+            """
+            The product of the dimensions of the shape must equal the number of scalars: 
```

Comment:
To make this message be one line.
```suggestion
            The product of the dimensions of the shape must equal the number of scalars: \
```